wagtail/models/i18n.py

```diff
@@ -210,6 +220,26 @@ def check(cls, **kwargs):
                 )
             )
 
+        if is_translation_model and not any(
```

Comment:
We need to combine this check with the above, so that the error is only raised if you don't have the `UniqueConstraint` nor the `unique_together`.

Currently, with this PR, the `UniqueConstraint` error will be shown if you have `unique_together`, and the `unique_together` error will be shown if you have `UniqueConstraint`. And you definitely shouldn't have both, so there's no way to get rid of the error.

And since `UniqueConstraint` is the way to go in the future, we should start by updating the error message to use it for the example.

As for whether we want to raise a warning/info when `unique_together` is used instead of `UniqueConstraint`, I feel like we should leave it out for now and revisit it in the future. Django hasn't officially deprecated it and [they still use it internally](https://github.com/django/django/blob/656192c2c96bb955a399d92f381e38fe2254fe17/django/contrib/auth/models.py#L75). I'd say we should wait and follow whatever Django will do in the future.